### ISSUE TITLE
Don't rely on codec

### ISSUE THREAD

**Issue by wangkuiyi** [sql-machine-learning/elasticdl#757]
The design and use of https://github.com/wangkuiyi/elasticdl/tree/develop/elasticdl/python/data/codec introduced the restriction that all training instances, consider image-label pairs, must have the same dimension, i.e., all images must have the same dimension and all instances must have the same number of labels.

It seems that by allowing ElasticDL users providing a function 

```python
def input_fn(record):
    ''' Convert a record (string) into the model input as a Python dictionary.
        Consider that the first layer of the model is tf.keras.DenseFeatures,
        whose constructor needs a list of feature column API calls, that converts 
        the output dictionary into tensors.
    '''
```

we should be able to remove the codec mechanism and get rid of the restriction.

What do you think? @skydoorkai @zhujl1991 @yuyicg @zou000 ?
